modules/phase_field/src/kernels/ACInterface.C

```diff
@@ -9,29 +9,60 @@
 template<>
 InputParameters validParams<ACInterface>()
 {
-  InputParameters params = validParams<KernelGrad>();
+  InputParameters params = validParams<Kernel>();
   params.addClassDescription("Gradient energy Allen-Cahn Kernel");
   params.addParam<MaterialPropertyName>("mob_name", "L", "The mobility used with the kernel");
   params.addParam<MaterialPropertyName>("kappa_name", "kappa_op", "The kappa used with the kernel");
   params.addCoupledVar("args", "Vector of nonlinear variable arguments this object depends on");
+
+  params.addParam<bool>("variable_L", true, "The mobility is a function of any non-linear variable");
+  params.addParam<bool>("variable_kappa", false, "Kappa is a function of any non-linear variable (must use ACInterfaceKappa Kernel along with this option)");
+
   return params;
 }
 
 ACInterface::ACInterface(const InputParameters & parameters) :
-    DerivativeMaterialInterface<JvarMapInterface<KernelGrad> >(parameters),
-    _kappa(getMaterialProperty<Real>("kappa_name")),
+    DerivativeMaterialInterface<JvarMapInterface<Kernel> >(parameters),
     _L(getMaterialProperty<Real>("mob_name")),
-    _dLdop(getMaterialPropertyDerivative<Real>("mob_name", _var.name()))
+    _kappa(getMaterialProperty<Real>("kappa_name")),
+    _variable_L(getParam<bool>("variable_L")),
+    _variable_kappa(getParam<bool>("variable_kappa")),
+    _dLdop(getMaterialPropertyDerivative<Real>("mob_name", _var.name())),
+    _d2Ldop2(getMaterialPropertyDerivative<Real>("mob_name", _var.name(), _var.name())),
+    _dkappadop(getMaterialPropertyDerivative<Real>("kappa_name", _var.name())),
+    _d2kappadop2(getMaterialPropertyDerivative<Real>("kappa_name", _var.name(), _var.name())),
+    _nvar(_coupled_moose_vars.size()),
+    _dLdarg(_nvar),
+    _d2Ldargdop(_nvar),
+    _d2Ldarg2(_nvar),
+    _dkappadarg(_nvar),
+    _d2kappadargdop(_nvar),
+    _d2kappadarg2(_nvar),
+    _gradarg(_nvar)
 {
-  // Get number of coupled variables
-  unsigned int nvar = _coupled_moose_vars.size();
+  // Get mobility and kappa derivatives and coupled variable gradients
+  for (unsigned int i = 0; i < _nvar; ++i)
+  {
+    MooseVariable *ivar = _coupled_moose_vars[i];
+
+    _dLdarg[i] = &getMaterialPropertyDerivative<Real>("mob_name", ivar->name());
```

Comment:
No. The validation needs to be in ```ACInterface::initialSetup()```, where it already is! **But** we need to make the ```validateCoupling``` to make sure that we _also_ validate that all AuxVariables are coupled (they could affect the gradient of L or kappa!).

I will make that change.